src/data_snack/snack.py

```diff
@@ -101,7 +100,7 @@ def _get_compound(self, cls: Type[CompoundEntity], key_values: List[Any]) -> Opt
         source_entities = []
         for source in cls.Meta.sources:
```

Comment:
This is a great place to add parallelization in the future. Can you add a `# TODO: parallelize processing for multiple sources